engine/consensus/compliance/engine.go

```diff
@@ -17,32 +18,36 @@ import (
 	"github.com/onflow/flow-go/module/irrecoverable"
 	"github.com/onflow/flow-go/module/metrics"
 	"github.com/onflow/flow-go/module/util"
-	"github.com/onflow/flow-go/network"
-	"github.com/onflow/flow-go/network/channels"
 	"github.com/onflow/flow-go/state/protocol"
 	"github.com/onflow/flow-go/storage"
 )
 
+// helper type used to pass originID and block through FIFO queue
+type inboundBlock struct {
+	originID flow.Identifier
+	block    *messages.BlockProposal
+}
```

Comment:
One option for addressing the proposerID/originID mismatch issue: We could make this a public type like `SyncedBlock` to capture the originID between components.